pele_platform/Adaptive/equilibrator.py

```diff
@@ -0,0 +1,146 @@
+            Cluster values from Parameters object (will contain default values if the user did not set any).
+        """
+        self.parameters = parameters
+        self.pele_dir = pele_dir
+        self.equilibration_parameters = None
+        self.preprocessed_system = preprocessed_system
+        self.cluster_values = cluster_values
+
+    def run(self):
+        """
+        Runs the equilibration workflow to extract cluster values based on protein-ligand contacts and adjust the main
+        simulation parameters accordingly.
+        """
+        self.generate_equilibration_parameters()
+        self.run_equilibration()
+
+        cluster_conditions = self.extract_cluster_conditions(
+            os.path.join(
+                self.equilibration_parameters.pele_dir,
+                self.equilibration_parameters.output,
+            )
+        )
+
+        return cluster_conditions
+
+    def generate_equilibration_parameters(self):
+        """
+        Generates parameters for the short equilibration simulation.
+        """
+        equilibration_parameters = deepcopy(self.parameters)
+        equilibration_parameters.iterations = 1
+        equilibration_parameters.pele_steps = 5
+        equilibration_parameters.cluster_conditions = None
+        equilibration_parameters.folder = os.path.join(
+            self.pele_dir, "QuickEquilibration"
+        )
+        equilibration_parameters.analyse = False
+        equilibration_parameters.no_ppp = True
+        equilibration_parameters.system = self.preprocessed_system
+
+        self.equilibration_parameters = equilibration_parameters
+
+    def run_equilibration(self):
+        """
+        Launches the simulation with equilibration parameters.
+        """
+        from pele_platform.Adaptive.simulation import run_adaptive
+
+        self.equilibration_parameters = run_adaptive(self.equilibration_parameters)
+
+    def extract_cluster_conditions(self, output_path):
+        """
+        Extracts contacts and thresholds identified by AdaptivePELE to select the best values for Adaptive clustering.
+
+        Returns
+        -------
+        cluster_conditions : List[float]
+            List of cluster conditions.
+        """
+        clustering_file = os.path.join(output_path, "0", "clustering", "summary.txt")
+        n_cluster_values = len(eval(self.cluster_values))   # cluster_values are a string
+
+        if not os.path.exists(clustering_file):
+            raise FileNotFoundError(
+                f"Could not locate {clustering_file} necessary to extract cluster values."
+            )
+
+        with open(clustering_file, "r") as file:
+            df = pd.read_csv(file, delimiter=r"\s+")
+        contacts = df["contacts"].tolist()
+
+        # To avoid cases where there is only one cluster and linspace will fail
+        if min(contacts) < max(contacts):
+            minimum = min(contacts)
+        else:
+            minimum = max(contacts) - 1
```

Comment:
I would change this line for a safer one: `minimum = max(max(contacts) - 1, 0)`. We must ensure it to be non negative